src/edit/index.vue

```diff
@@ -1,9 +1,138 @@
+
+const DEFAULT_ERROR_MSG = 'Something went wrong.';
+const CANCEL_UPLOAD_ERROR_MSG = 'Upload canceled by leaving the page.';
+const UPLOADING_MSG = 'Video is uploading... Do not leave the page.';
+const PROCESSING_MSG = 'Video is processing...';
+
 export default {
-  name: 'tce-api-video'
+  name: 'tce-api-video',
+  inject: ['$elementBus'],
+  props: {
+    element: { type: Object, required: true },
+    isFocused: { type: Boolean, default: false },
+    isDisabled: { type: Boolean, default: false }
+  },
+  data: () => ({ file: null }),
+  computed: {
+    videoId: ({ element }) => element.data?.videoId,
+    uploadUrl: ({ element }) => element.data?.uploadUrl,
+    embedCode: ({ element }) => element.data?.embedCode,
+    fileName: ({ element }) => element.data?.fileName,
+    playable: ({ element }) => element.data?.playable,
+    status: ({ element }) => element.data?.status,
+    error: ({ element }) => element.data?.error,
+    showError: ({ status }) => status === ELEMENT_STATE.ERROR,
+    showPlaceholder: ({ error, fileName }) => !error && !fileName,
+    infoMessage: ({ error, status, playable }) => {
+      if (error) return;
+      if (status === ELEMENT_STATE.UPLOADING) return UPLOADING_MSG;
+      if (!playable) return PROCESSING_MSG;
+    }
+  },
+  methods: {
+    appendVideo() {
+      const { player } = this.$refs;
+      if (!player) return;
+      player.innerHTML = this.embedCode;
+    },
+    checkIfError() {
+      const { status, file, element } = this;
+      if (status !== ELEMENT_STATE.UPLOADING || file) return;
+      this.$emit('save', {
+        ...element.data,
+        status: ELEMENT_STATE.ERROR,
+        error: CANCEL_UPLOAD_ERROR_MSG
+      });
+    }
+  },
+  watch: {
+    embedCode: 'appendVideo',
+    videoId() {
+      const { videoId, file, uploadUrl: url } = this;
+      if (!videoId || !file || !url) return;
+      return upload({ url, file, videoId })
+        .then(() => {
+          this.file = null;
+          this.$emit('save', { ...this.element.data, status: ELEMENT_STATE.UPLOADED });
+        })
+        .catch(err => this.$elementBus.emit('error', err.response));
+    }
+  },
+  mounted() {
+    this.checkIfError();
+    this.appendVideo();
+
+    this.$elementBus.on('save', ({ file }) => {
+      this.file = file;
+      this.$emit('save', { fileName: file.name, status: ELEMENT_STATE.UPLOADING });
+    });
+
+    this.$elementBus.on('error', ({ data }) => {
+      this.$emit('save', {
+        ...this.element.data,
+        status: ELEMENT_STATE.ERROR,
+        error: get(data, 'error.message', DEFAULT_ERROR_MSG)
+      });
+    });
+  },
+  components: { ElementPlaceholder }
 };
 </script>
+
+<style lang="scss" scoped>
+.tce-api-video {
+  position: relative;
+}
+
+.overlay {
+  position: absolute;
+  z-index: 3;
+  width: 100%;
+  height: 100%;
+  background: rgba(16, 16, 16, 0.85);
+
+  .message {
+    position: relative;
+    top: 45%;
+    font-size: 1.125rem !important;
```

Comment:
What's overriding this?

Reply:
Seems to be nothing 😅 I copied this from the `tce-video` without checking it out, my bad 😐